thin-replica-server/test/thin_replica_server_test.cpp

```diff
@@ -551,6 +863,29 @@ TEST(thin_replica_server_test, SubscribeWithWrongBlockId) {
   EXPECT_EQ(status.error_code(), grpc::StatusCode::FAILED_PRECONDITION);
 }
 
+TEST(thin_replica_server_test, SubscribeWithWrongEventGroupId) {
+  FakeStorage storage(generateEventGroupMap(0, 0));
+  auto live_update_event_groups = generateEventGroupMap(0, 0);
+  TestStateMachine<Data> state_machine{storage, live_update_event_groups, 1};
+  TestSubBufferList<Data> buffer{state_machine};
+  TestServerWriter<Data> stream{state_machine};
+
+  bool is_insecure_trs = true;
+  std::string tls_trs_cert_path;
+  std::unordered_set<std::string> client_id_set;
+  uint16_t update_metrics_aggregator_thresh = 100;
+
+  TestServerContext context;
+  auto trs_config = std::make_unique<concord::thin_replica::ThinReplicaServerConfig>(
+      is_insecure_trs, tls_trs_cert_path, &storage, buffer, client_id_set, update_metrics_aggregator_thresh);
+  concord::thin_replica::ThinReplicaImpl replica(std::move(trs_config), std::make_shared<concordMetrics::Aggregator>());
+  SubscriptionRequest request;
+  request.mutable_event_groups()->set_event_group_id(kLastEventGroupId + 100);
+  auto status =
+      replica.SubscribeToUpdates<TestServerContext, TestServerWriter<Data>, Data>(&context, &request, &stream);
+  EXPECT_EQ(status.error_code(), grpc::StatusCode::FAILED_PRECONDITION);
```

Comment:
FYI, I was recently talking with @f-squirrel that we might want to change this behavior. If the load is high then it might be reasonable to wait instead of cancelling the request. In any case, the TRC can choose to cancel at any point in time. This error code can maybe be used for pruned data.

Reply:
Will add in a separate PR.